src/os/data/layertreesearch.js

```diff
@@ -57,7 +57,7 @@ class LayerTreeSearch extends AbstractGroupByTreeSearch {
     var layer = /** @type {!ILayer} */ (item);
     var node;
 
-    if (layer.getTreeNode) {
+    if (typeof /** @type {!ITreeNodeSupplier} */ (layer).getTreeNode === 'function') {
```

Comment:
Might be annoying to add all the cases as implementers of `ITreeNodeSupplier`, but it would be less gross to do an `implements` check here.

Reply:
Yeah I didn't want to add a breaking change to fix a typing issue from the original code. The `typeof` check just made the original check a little more specific, but also shouldn't be necessary.